rasa/train.py

```diff
@@ -163,31 +184,43 @@ def train_core(
 
     _train_path = train_path or tempfile.mkdtemp()
 
+    story_directory = data.get_core_directory(stories)
+
+    if not os.listdir(story_directory):
+        print_error(
+            "No dialogue data given. Please provide dialogue data in order to "
+            "train a Rasa Core model."
+        )
+        return
+
     # normal (not compare) training
-    core_model = await rasa.core.train(
+    await rasa.core.train(
         domain_file=domain,
-        stories_file=data.get_core_directory(stories),
+        stories_file=story_directory,
         output_path=os.path.join(_train_path, "core"),
         policy_config=config,
         kwargs=kwargs,
     )
 
     if not train_path:
         # Only Core was trained.
-        stories = data.get_core_directory(stories)
         output_path = create_output_path(output, prefix="core-")
-        new_fingerprint = model.model_fingerprint(config, domain, stories=stories)
+        new_fingerprint = model.model_fingerprint(
+            config, domain, stories=story_directory
+        )
         model.create_package_rasa(_train_path, output_path, new_fingerprint)
         print_success(
             "Your Rasa Core model is trained and saved at '{}'.".format(output_path)
         )
 
-    return core_model
+        return output_path
+
+    return _train_path
 
 
 def train_nlu(
     config: Text, nlu_data: Text, output: Text, train_path: Optional[Text]
-) -> Optional["Interpreter"]:
+) -> Optional[Text]:
```

Comment:
You can also remove these lines then (type checking import is not needed anymore)
```
if typing.TYPE_CHECKING:
    from rasa.nlu.model import Interpreter
```